app/src/main/java/com/nutomic/syncthingandroid/service/RunConditionMonitor.java

```diff
@@ -0,0 +1,310 @@
+        if (prefRunOnlyWhenCharging && !isCharging()) {
+            Log.v(TAG, "decideShouldRun: prefRunOnlyWhenCharging && !isCharging");
+            return false;
+        }
+
+        // Run only on wifi.
+        if (prefRunOnlyOnWifi && !isWifiOrEthernetConnection()) {
+            // Not on wifi.
+            Log.v(TAG, "decideShouldRun: prefRunOnlyOnWifi && !isWifiOrEthernetConnection");
+            return false;
+        }
+
+        // Run only on whitelisted wifi ssids.
+        if (prefRunOnlyOnWifi && prefWifiWhitelistEnabled) {
+            // Wifi connection detected. Wifi ssid whitelist enabled.
+            Log.v(TAG, "decideShouldRun: prefRunOnlyOnWifi && prefWifiWhitelistEnabled");
+            return isWifiConnectionWhitelisted(whitelistedWifiSsids);
+        }
+
+        /**
+         * Respect power saving, device is not in power-save mode.
+         * Always run in background is the only pref that is enabled.
+         * Run only when charging, charging.
+         * Run only on wifi, wifi connection detected, wifi ssid whitelist disabled.
+         */
+        Log.v(TAG, "decideShouldRun: return true");
+        return true;
+    }
+
+    /**
+     * Functions for run condition information retrieval.
+     */
+    private boolean isCharging() {
+        Intent batteryIntent = mContext.registerReceiver(null, new IntentFilter(Intent.ACTION_BATTERY_CHANGED));
+        int status = batteryIntent.getIntExtra(BatteryManager.EXTRA_STATUS, -1);
+        return status == BatteryManager.BATTERY_STATUS_CHARGING ||
+            status == BatteryManager.BATTERY_STATUS_FULL;
+    }
+
+    @TargetApi(21)
+    private boolean isPowerSaving() {
+        if (Build.VERSION.SDK_INT < Build.VERSION_CODES.LOLLIPOP) {
+            Log.e(TAG, "isPowerSaving may not be called on pre-lollipop android versions.");
+            return false;
+        }
+        PowerManager powerManager = (PowerManager) mContext.getSystemService(Context.POWER_SERVICE);
+        if (powerManager == null) {
+            Log.e(TAG, "getSystemService(POWER_SERVICE) unexpectedly returned NULL.");
+            return false;
+        }
+        return powerManager.isPowerSaveMode();
+    }
+
+    private boolean isFlightMode() {
+        ConnectivityManager cm = (ConnectivityManager) mContext.getSystemService(Context.CONNECTIVITY_SERVICE);
+        NetworkInfo ni = cm.getActiveNetworkInfo();
+        return ni == null;
+    }
+
+    private boolean isMeteredNetworkConnection() {
+        ConnectivityManager cm = (ConnectivityManager) mContext.getSystemService(Context.CONNECTIVITY_SERVICE);
+        NetworkInfo ni = cm.getActiveNetworkInfo();
+        if (ni == null) {
+            // In flight mode.
+            return false;
+        }
+        if (!ni.isConnected()) {
+            // No network connection.
+            return false;
+        }
+        return cm.isActiveNetworkMetered();
+    }
+
+    private boolean isMobileDataConnection() {
```

Comment:
Not used?

Reply:
Currently yes. Is it okay if I'll need that function later as I intend to make a new rjn condition gui proposal in a follow-up pr?

Reply:
I think that's when you should add these functions, in a PR where they are actually used.

Reply:
Ok, removed them. Ready?